### ISSUE TITLE
Microservices proposal: CHE should be able to see/route request to a desired environment

### ISSUE THREAD

**Issue by rafabene** [openshiftio/openshift.io#1017]
Use case: Developer is working on a microservices A inside CHE. The application needs to performa a call to microservices B and C that are deployed in a Stage enviroment. CHE should allow these requests to be routed to an specific environment (Stage or Run).

**Comment by l0rd** [comment]
In practice that would mean creating a service in the Che namespace and a route in the Stage namespace. That should be done dynamically (using https://github.com/fabric8io/exposecontroller?). 

@jstrachan I remember you were talking about that a few weeks ago on openshiftio-saas channel: any thoughts about that?

**Comment by jstrachan** [comment]
yeah, we should be able to create a little CLI tool to do things like this; mirror services from Stage/Run into the Che namespace. Its a generally useful thing to do
https://github.com/openshift/origin/issues/15980
